### PR TITLE
[React] change createElement declaration

### PR DESCRIPTION
**Pull request by plantain-00**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:  https://github.com/Microsoft/TypeScript/issues/15019
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by plantain-00** [comment]
The failure comes from:
`The following packages had errors: react-app, react-dom, react-transition-group`
Am I expected to change these packages to make CI pass?

File: types/react/v15/index.d.ts [hunk 1]
```diff
@@ -223,7 +223,7 @@ declare namespace React {
         props?: ClassAttributes<T> & P,
         ...children: ReactNode[]): CElement<P, T>;
     function createElement<P>(
-        type: ComponentClass<P>,
+        type: SFC<P> | ComponentClass<P>,
```
Comment (vsiao) [inline types/react/v15/index.d.ts]: seems reasonable to have this catch-all case also take `string`. Do you mind adding that?

Comment (plantain-00) [reply types/react/v15/index.d.ts]: Just added string.

**Comment by plantain-00** [comment]
@andy-ms There are some `no-unnecessary-generics` errors of other libs in the CI, can you show me how to remove the `no-unnecessary-generics` errors?
```ts
// react-dom
export function unstable_renderSubtreeIntoContainer<P, T extends Component<P, ComponentState>>(
    parentComponent: Component<any>,
    element: CElement<P, T>, // <- P
    container: Element,
    callback?: (component: T) => any): T;
```

**Comment by andy-ms** [comment]
I've done #19906, #19907, and  #19908 which should help. Please merge from the latest master once those are all in.

**Review by Passhag — APPROVED** [review]

**Pull request merged**